rust/src/storage/azure/mod.rs

```diff
@@ -368,7 +368,7 @@ impl StorageBackend for AdlsGen2Backend {
             .flat_map(|it| match it {
                 Ok(paths) => Either::Left(stream::iter(paths.into_iter().map(|p| {
                     Ok(ObjectMeta {
-                        path: path.to_string(),
+                        path: format!("adls2://{}/{}", self.file_system_name, path.to_owned()),
```

Comment:
It's unclear to me how this fixes the bug reported:

> The file system and Azure list_objs implementations return only objects within the directory specified by path which differs with the S3 implementation which returns all objects that have a prefix of path. 

It seems like this change just changes the format of the file paths in `list_dir`? I can clearly see `.recursive(false)` above, so maybe that's the change being asked for? Am I missing something?

Reply:
you are eight, i was too hasty there. .. but this yet another bug....

Reply:
btw. I do remember now where this comes from - likely the same for the local store. SO far (aside from vacuum) we never list the file directory, bur always have the delta log when accessing files. Vacuum assumes that there might be orphaned files, not discoverable via the delta log, so it does a list on the files. This is where folders first come into play ... in an ideal world this would not even be necessary, as we should only remove tombstones if we remove files - right?

Reply:
I think the reason we list is there might be some files left over from aborted transactions, which wouldn't show up in the logs.

Reply:
Since we need to fix local as well, I was thinking to move the walkdir based implementation over from object_store_rs ... also really looking forward to having this migration :). I think we need some more more on the auth mechanism over there before we can switch, but getting there ...

Reply:
Yes, I'll be focusing on that soon.